### PR TITLE
Cherrypicker: assign issue on cherrypick error

### PR DESCRIPTION
**Pull request by clarketm**
fix https://github.com/kubernetes/test-infra/issues/16487
fix https://github.com/istio/test-infra/issues/2423

### CODE REVIEW

**Comment by fejta** [comment]
I generally ignore WIP PRs, do you want feedback here or are you just trying to see if presubmits pass?

**Comment by clarketm** [comment]
> I generally ignore WIP PRs, do you want feedback here or are you just trying to see if presubmits pass?

No need to review yet, I am just checking presubmits as you guessed. I am still writing tests and verifying e2e.

File: prow/external-plugins/cherrypicker/main.go [hunk 1]
```diff
@@ -65,6 +66,7 @@ func gatherOptions() options {
 	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
 	fs.BoolVar(&o.prowAssignments, "use-prow-assignments", true, "Use prow commands to assign cherrypicked PRs.")
 	fs.BoolVar(&o.allowAll, "allow-all", false, "Allow anybody to use automated cherrypicks by skipping GitHub organization membership checks.")
+	fs.BoolVar(&o.issueOnFailure, "create-issue-on-failure", false, "Create a GitHub issue and assign it to the requestor if a cherrypick fails.")
```
Comment (kargakis) [inline prow/external-plugins/cherrypicker/main.go:69]: There are other reasons why a cherrypick can fail (eg. flaky Github API) so maybe make it clearer here that this is specifically for conflicts in the cherrypick?

Comment (clarketm) [reply prow/external-plugins/cherrypicker/main.go]: Makes sense. I renamed this flag to `--create-issue-on-conflict` and updated the help text accordingly.

**Comment by kargakis** [comment]
Seems there is a conflict that needs to be resolved.

**Comment by kargakis** [comment]
/lgtm

@fejta needs an approval

**Pull request merged**